config/dune

```diff
@@ -0,0 +1,12 @@
+(library
+ (name config)
+ (synopsis "Coq Configuration Variables")
+ (public_name coq.config)
+ (wrapped false))
+
+; We provide a fast developer profile by default
+(rule
+ (targets coq_config.ml)
+ (mode fallback)
+ (deps %{project_root}/configure.ml %{project_root}/dev/ocamldebug-coq.run)
+ (action (chdir %{project_root} (run %{ocaml} configure.ml -local -warn-error yes -native-compiler yes))))
```

Comment:
It doesn't really matter as this will go away in the next Dune-related PR that will remove dependency from configure. Anyways, neither option to configure has an effect on Dune as of today. Dune has its own notion of profile by the way.

Reply:
Yes, but let's try to keep the two in sync until configure is removed as a dependency.

Reply:
It won't? OK, maybe remove these arguments then and add a comment saying that they don't matter... Not a big deal but it would be clearer I guess.

Reply:
-local and -native-compiler have an effect on coq_config don't they?

Reply:
`-native-compiler` effect is not seen as Dune still don't pass the native flag to the stdlib [due to some linking stuff, in particular related to the "hidden" native compiler library, more to follow on a different PR]

`-local` is indeed used and Dune relies on it; my next PR will hopefully remove `-local` altogether.

Reply:
Note that this PR doesn't fix the mess we have with `-local` so indeed it is not good at all from this point of view, for now the Dune build is only usable for developers and will require more work to properly support hygienic builds. I think tho the tradeoff is in favor of merging as already it provides quite a few benefits over make [for example to instrument code], but yeah maybe you would like to be better here, I dunno.

Reply:
For instance, `dune exec coqtop` and `dune exec coqtop.byte` do differ in funny ways :S How bad the mess can be...